### PR TITLE
Add font-lock to arrows

### PR DESCRIPTION
**Pull request by mpanarin**
Big `with` constructs and function pipelines are hard to follow, because arrows are not highlighted in any form. 
This adds font-locking for them

### CODE REVIEW

**Comment by victorolinasc** [comment]
This is quite interesting. I am for this contribution but I am not 100% sure the best face here would be `keyword`. These "arrows" are certainly part of the syntax and should be highlighted but they are not exactly "keywords" I think. 

In any case, could you please add a test for this case? We can discuss the face later on.

Thanks for your contribution :)

**Comment by jsmestad** [comment]
@victorolinasc we should look at what the elixir team is putting into tree-sitter elixir and just copy that :100:

**Review by jsmestad — APPROVED** [review]
